packages/client/docusaurus/docs/javascript/02-tutorials/02-audio-room.mdx

```diff
@@ -0,0 +1,571 @@
+
+  // allow button click handler
+  allowButton.addEventListener('click', async () => {
+    await call.grantPermissions(userId, permissions);
+    permissionRequestListElement!.removeChild(requestElement);
+    permissionRequestsElement!.style.display = permissionRequestListElement!
+      .childElementCount
+      ? 'block'
+      : 'none';
+  });
+
+  // deny button click handler
+  denyButton.addEventListener('click', async () => {
+    await call.revokePermissions(userId, permissions);
+    permissionRequestListElement!.removeChild(requestElement);
+    permissionRequestsElement!.style.display = permissionRequestListElement!
+      .childElementCount
+      ? 'block'
+      : 'none';
+  });
+
+  // bridge everything together
+  requestElement.appendChild(requestTextElement);
+  requestElement.appendChild(allowButton);
+  requestElement.appendChild(denyButton);
+  permissionRequestListElement!.appendChild(requestElement);
+  permissionRequestsElement!.style.display = 'block';
+});
+```
+
+### Step 6 - Group participants
+
+It is common for audio rooms and similar interactive audio/video experiences to show users in separate groups.
+Let's see how we can update this application to render participants in two separate sections: _Speakers_ and _Listeners_.
+
+Building custom layout is straight forward. All we need to do is to apply some filtering to the result of `call.state.participants$` observable.
+
+Lets update our _Participant Panel_ section:
+
+```html title=index.html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>Audio Rooms Tutorial</title>
+  </head>
+  <body>
+    <div id="app">
+      <div class="permission-requests" style="display: none">
+        <h4>Permission requests</h4>
+        <ul class="permission-requests-list">
+          <!-- we'll add the permission requests here -->
+        </ul>
+      </div>
+      <div class="description-panel">
+        <h2 class="title"></h2>
+        <h3 class="description"></h3>
+        <p class="participant-count"></p>
+      </div>
+      // highlight-start
+      <div class="participants-panel">
+        <h4>Speakers</h4>
+        <ul class="speakers-list">
+          <!-- we'll add the participants here -->
+        </ul>
+        <h4>Listeners</h4>
+        <ul class="listeners-list">
+          <!-- we'll add the participants here -->
+        </ul>
+      </div>
+      // highlight-end
+      <div class="controls-panel">
+        <button type="button" class="toggle-live-button">Go Live</button>
+        <button type="button" class="toggle-mute-button">Mute</button>
+      </div>
+    </div>
+    <script type="module" src="/src/main.ts"></script>
+  </body>
+</html>
+```
+
+Now, let's update the code accordingly:
+
+```tsx title=src/main.ts
+import { SfuModels, StreamVideoParticipant } from '@stream-io/video-client';
+
+// ... previous code
+
+// participant lists
+const participantListElement = $$('.participants-panel .speakers-list');
```

Comment:
```suggestion
const speakerListElement = $$('.participants-panel .speakers-list');
```

Reply:
This rename indicates some other changes in the code as well

Reply:
Fixed with: https://github.com/GetStream/stream-video-js/pull/1190/commits/e4a73f62f265bb752fc0d231897fbee8becf6f97